### PR TITLE
1 register new user

### LINKED ISSUE #1: Create User model that validates email address as (1) having authorized domain name

### LINKED ISSUE #2: Make users confirmable

### LINKED ISSUE #3: Make users registerable

### PR DESCRIPTION
**Pull request by dcox**
Resolve #1 with custom email validation to check for the presence of a company email address.
Resolve #2 by allowing users to sign in.
Resolve #3 by allowing users to reset a forgotten password.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class ApplicationController < ActionController::Base
```
Comment (eliotsykes) [inline app/controllers/application_controller.rb:1]: @dcox Consider adding a `before_action :authenticate_user!` to `ApplicationController` to enforce authentication by default for all controller actions.

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` on Rails 4+ applications as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = '0d3a2b26cb0737a8b4cc0b6852aeffe3a7d9062552e7de00f92ad42ad00fa5461f39f9f7abf232a5d77b1e6befdd8756c85da35f0a4e699b9087dc44ae4e945d'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
Comment (eliotsykes) [inline config/initializers/devise.rb:15]: @dcox consider customizing sent-from address for devise mails.

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
+Rails.application.routes.draw do
+  get 'users/create'
+
+  get 'users/new'
+
```
Comment (eliotsykes) [inline config/routes.rb:5]: Review `get` routes on line 2 and 4 - they may have been added by a Rails generator but may not be needed. If they are needed, prefer using `resources :users, only: [...]` instead as the `resources ...`-routes tend to be a bit easier to work with than the `get ...`-routes (for example the `*_path` helper naming tends to be better with `resources ...`-routes).

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (dcox) [inline app/controllers/application_controller.rb:1]: 👍

**Pull request merged**